java/src/org/openqa/selenium/grid/config/MapConfig.java

```diff
@@ -65,7 +68,24 @@ public Optional<List<String>> getAll(String section, String option) {
     }
 
     Object value = rawSection.get(option);
-    return value == null ? Optional.empty() : Optional.of(ImmutableList.of(String.valueOf(value)));
+    if (value == null) {
+      return Optional.empty();
+    }
+
+    if (value instanceof Collection<?>) {
+      Collection<?> collection = (Collection<?>) value;
+      // Case when an array of map is used as config
+      if (collection.stream().anyMatch(item -> item instanceof Map)) {
+        List<String> toReturn = collection.stream()
```

Comment:
If the assumed structure is section/option/value, don't the values need to be encoded as JSON to avoid loss of fidelity?